lib/kramdown/options.rb

```diff
@@ -388,9 +392,13 @@ def self.simple_hash_validator(val, name)
       Default: lsquo,rsquo,ldquo,rdquo
       Used by: HTML/Latex converter
     EOF
-      val = simple_array_validator(val, :smart_quotes, 4)
-      val.map! {|v| Integer(v) rescue v }
-      val
+      if val.eql?(SMART_QUOTES_CSV) || val.eql?(SMART_QUOTES_ENTITIES)
```

Comment:
Why use `eql?`? Shouldn't it better be `==`?

Reply:
I used `Array#eql?` because the [docs](https://ruby-doc.org/core-2.4.0/Array.html#method-i-eql-3F) says that is based on [`Object#eql?`](https://ruby-doc.org/core-2.4.0/Object.html#method-i-eql-3F) which compares the `hash` value of the objects.

But tests show that results are the same with `#==` and `#eql?` :man_shrugging:

Reply:
I think we want `==` here. As far as I know, `eql?` should only be used when hashing is concerned.

Reply:
Okay. The branch has been updated.